### ISSUE TITLE
TypeError: server.changed is not a function

### ISSUE THREAD

**Issue by CiceroBjorn** [toddmotto/html5blank#159]
This happens when I save a file? 

`cicerobjorn@CDGbootstrap01:/var/www/html/wp-content/themes/html5blank$ gulp watch
[12:18:17] Using gulpfile /var/www/html/wp-content/themes/html5blank/gulpfile.js
[12:18:17] Starting 'template'...
`template` task run in `development` environment
[12:18:17] Starting 'sass'...
[12:18:17] Starting 'jshint'...
[12:18:18] Finished 'jshint' after 234 ms
[12:18:18] Finished 'template' after 371 ms
[12:18:18] Finished 'sass' after 316 ms
[12:18:18] Starting 'styles'...
`styles` task run in `development` environment
[12:18:18] Finished 'styles' after 259 ms
[12:18:18] Starting 'watch'...
[12:18:18] Finished 'watch' after 109 ms
/var/www/html/wp-content/themes/html5blank/src/header.php
/var/www/html/wp-content/themes/html5blank/gulpfile.js:147
		server.changed( file.path );
		       ^

TypeError: server.changed is not a function
    at EventEmitter.<anonymous> (/var/www/html/wp-content/themes/html5blank/gulpfile.js:147:10)
    at emitOne (events.js:77:13)
    at EventEmitter.emit (events.js:169:7)
    at Gaze.<anonymous> (/var/www/html/wp-content/themes/html5blank/node_modules/glob-watcher/index.js:17:11)
    at emitTwo (events.js:87:13)
    at Gaze.emit (events.js:172:7)
    at Gaze.emit (/var/www/html/wp-content/themes/html5blank/node_modules/gaze/lib/gaze.js:129:32)
    at /var/www/html/wp-content/themes/html5blank/node_modules/gaze/lib/gaze.js:415:16
    at StatWatcher._pollers.(anonymous function) (/var/www/html/wp-content/themes/html5blank/node_modules/gaze/lib/gaze.js:326:7)
    at emitTwo (events.js:87:13)`

**Comment by oriSomething** [comment]
2 options seems to me possible:
- You update Node version or partly node packages
- LiveReload plugin related

1. What is your `Node`'s version?
2. If you didn't already do that, please go to the root of the project and try running:

```
$ rm -rf node_modules
$ npm i
```

By the way, I understand it's Linux?

**Comment by CiceroBjorn** [comment]
Yes, sorry, it's a Virtual Box VM with Ubuntu 16.04 installed.
Node v4.2.6

```
$rm -rf node_modules
$npm i
```
Did not help anything unfortunately.

I comment out server.changed in the gulpfile.js and the gulp watch function is running, but there is still no LiveReload..

```
/** Watch for livereoad */
	gulp.watch([
		"src/js/**/*.js",
		"src/*.php",
		"src/*.css"
	]).on( "change", function( file ) {
		console.log( file.path );
		//server.changed( file.path );
	}); 
```

**Comment by oriSomething** [comment]
I can confirm this bug. The docs, didn't show any changes in the API. I'm on it…

**Comment by oriSomething** [comment]
Can you please try the fix before I merge https://github.com/toddmotto/html5blank/pull/161 ?

**Comment by CiceroBjorn** [comment]
That sort of fixed the bug, the `gulp watch` function now compiles the css etc. properly. 
But I still have no LiveReloading, I still have to manually refresh the webpage to see the changes? 

I installed [BrowerSync](https://browsersync.io/) instead and configured it like this:

```
/** Livereload */
gulp.task( "watch", [ "browserSync", "template", "styles", "jshint" ], function() {
	var server = $.livereload();

	/** Watch for livereoad */
	gulp.watch("src/js/**/*.js",browserSync.reload);
	gulp.watch("src/*.php", browserSync.reload);
	gulp.watch("src/*.css", browserSync.reload);

```

That works for me.

**Comment by oriSomething** [comment]
Okay, It's probably because you need the LiveReload plugin installed in the browser. I think I add the needed `<script>`s automatically, And than there will be no need for the plugin.
I don't like the idea of using `BrowserSync` since as far as I know it means using a proxy, which makes it super complicated.
I might get in to it during the weekend to make sure it works automatically with no setup.

**Comment by GuiHarrison** [comment]
I'm having this problem as well... Any help?

**Comment by davidpenuelab** [comment]
In my case, with the help of @GuiHarrison and after reading the documentation from [browsersync](https://www.browsersync.io/docs/gulp#gulp-sass-css), I manage to solve it. 

What my site does, is:
– reloads every time there is a change in a php, css, js, scss file.

I did it this way:
When there is change in a scss file, i call ['styles']
then i am watching for changes in the js, php, and css files. Since the ['styles'] modifies my css file, browrsync calles the reload.

This is my code:

**package.json i added**
added browser sync so i can use it in gulpfile
```
 "browser-sync": "^2.14.0"
```

**gulpfile.js**
added variable for browser sync
```
var browserSync = require("browser-sync").create();
```
methods changed:
```
/** CSS Preprocessors */
gulp.task( "sass", function () {
	console.error( "SASS");
	return gulp.src( "src/css/sass/style.scss" )
		.pipe( $.sourcemaps.init() )
		.pipe( $.sass() )
		.pipe( $.sourcemaps.write( "." ) )
		.on( "error", function( e ) {
			console.error( e );
		})
		.pipe( gulp.dest( "src/css" ) )
		.pipe(browserSync.stream());

});

/** Livereload */
gulp.task( "watch", [ "template", "styles", "jshint" ], function() {
	console.log( "watch");
	
	browserSync.init({
        proxy: "mydomail.com/wp/"
    });

    gulp.watch("src/css/sass/**/*.scss", ["styles"]);

	/** Watch for livereoad */
	gulp.watch("src/js/**/*.js",browserSync.reload);
	gulp.watch("src/*.php", browserSync.reload);
	gulp.watch("src/css/**/*.css", browserSync.reload);


});
```

**Comment by oriSomething** [comment]
Hey sorry, it was a small fix that have being waited for to long. But I've merged #161 
Currently the LiveReload plugin is needed, but I reconsider doing some changes so it will with BrowserSync. But we might move everything to webpack, so it won't be needed.
I'm close this issue for now, let me know if you think it should be still open. thanks
